src/helpers/transport.rs

```diff
@@ -0,0 +1,121 @@
+use crate::error::BoxError;
+use crate::helpers::commands::RingConfiguration;
+use crate::helpers::network::Network;
+use crate::helpers::{Command, HelperIdentity};
+#[cfg(test)]
+use crate::test_fixture::network::{InMemoryEndpoint, InMemoryNetwork};
+use async_trait::async_trait;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Command has been rejected by {identity:?}: {inner}")]
+    CommandRejected {
+        identity: HelperIdentity,
+        #[source]
+        inner: BoxError,
+    },
+}
+
+#[async_trait]
+pub trait Transport {
+    type AppLayer: Network;
+
+    async fn send(&self, command: Command<'_>) -> Result<(), Error>;
+
+    /// TODO: remove, it is not transport's job to create this.
+    fn app_layer(&self, ring: &RingConfiguration) -> Self::AppLayer;
+}
+
+#[cfg(test)]
```

Comment:
All the rest of this file is `#[cfg(test)]`.  Can this be put into a module to make it a little cleaner?  It might even go in the test fixture.